spec/integration/websocket_spec.lua

```diff
@@ -0,0 +1,100 @@
+---@diagnostic disable: undefined-global
+-- luacheck: globals describe it before_each after_each
+require("busted.runner")()
+local websocket = require("websocket")
+local ConnectionProperties = require("connection_properties")
+local log = require("remotelog")
+local config = require("config")
+config.configure_logging()
+local params<const> = config.get_connection_params()
+local local_websocket_url<const> = "wss://" .. params.host .. ":" .. params.port
+local ssl_negotion_failed_error<const> = string.format("E-EDL-1: Error connecting to '%s': 'Failed SSL negotation'",
+                                                       local_websocket_url)
+
+local function connect(properties, url)
+    url = url or local_websocket_url
+    properties = properties or {}
+    local connection_properties = ConnectionProperties:create(properties)
+    local sock = websocket.connect(url, connection_properties)
+    finally(function() sock:close() end)
+    return sock
+end
+
+local function assert_connect_successful(properties)
+    local sock = connect(properties)
+    assert.is_not_nil(sock)
+    assert.is_false(sock.closed)
+    local result = sock:send_raw('{"command": "login", "protocolVersion": 3}')
+    assert.matches('.*"status":"ok".*', result)
+end
+
+local function assert_connect_fails(properties, expected_error)
+    expected_error = expected_error or ssl_negotion_failed_error
+    assert.error(function() connect(properties) end, expected_error)
+end
+
+describe("Websocket", function()
+    it("connects with default properties", function() --
+        assert_connect_successful()
+    end)
+
+    it("connects with tls_verify=none", function() --
+        assert_connect_successful({tls_verify = "none"})
+    end)
+
+    describe("fails connection with tls_verify option", function()
+        for _, tls_verify_option in ipairs({"invalid", "", "peer", "client_once", "fail_if_no_peer_cert"}) do
+            it(string.format("value %q", tls_verify_option), function() --
+                assert_connect_fails({tls_verify = tls_verify_option})
```

Comment:
`assert_connect_fails()` contains a default error message. In fact, all tests result in the same error, so I wanted to avoid duplicating the error message.